### DISCUSSION TITLE
Open default application with Linux

### DISCUSSION THREAD

**Discussion by pierrehub2b** [airsdk/Adobe-Runtime-Support#2762] [Q&A]
Hello,

I try to use the function '**openWithDefaultApplication**' with a file object with the Harman Flex SDK Air application on Linux (build with captive runtime option).
Is there something I have to do to make it works or this function is useless with Linux ?

Thank you for your help
Pierre

**Comment by ajwfrost** [comment]
It looks like this functionality is dependent upon the type of Linux desktop being used... so the original Adobe code may be a little old/obsolete here now. Can I check what version/variant of Linux you're using? We can do a bit of testing/debugging to see if this needs tweaking.

It would also be good if you can let us know the type of file you are trying it with, and what the file extension is, so we can ensure we're reproducing the issue in the same way.

thanks

**Reply by pierrehub2b** [reply]
Hello Andrew (again :-) )

We use for testing Ubuntu 22.10 with Wayland windowing system (we can use another Linux desktop distribution)
We want to open a text file with the defaut text editor installed on the system
We want to open an url with the default installed browser
and finnaly we want to open the default file explorer on a folder
